core/src/main/java/com/datastax/oss/driver/api/core/metadata/Metadata.java

```diff
@@ -114,4 +114,10 @@
    */
   @NonNull
   Optional<TokenMap> getTokenMap();
+
+  /** The cluster name for this cluster. */
```

Comment:
"for this session" to be more accurate.

Please also explain that this corresponds to the value of `system.local.cluster_name` on the server, just to lift any ambiguity.

Finally, also mention that this method has a default implementation only for the sake of backward compatibility: in the unlikely case that someone implements the interface in their code (e.g. a wrapper), they are expected to override it.
(btw I agree with the decision to make it default, it's simpler than introducing yet-another-subinterface)